File: ruby/stack.rb
class Stack
  def initialize
    @items = []
  end

  def push(item)
    @items.push(item)
  end

  def pop
    @items.pop
  end

  def peek
    @items[@items.length -  1]
  end

  def all
    @items
  end
end

File: node/stack.js
class Stack {
  constructor() {
    this.items = [];
  }

  peek() {
    return this.items[this.items-length - 1];
  }

  all() {
    return this.items;
  }

  push(item) {
    this.items.push(item);
  }

  pop() {
    this.items.pop();
  }
}

module.exports = Stack;
